### PR TITLE
feat(api/web): Attachments 

### PR DESCRIPTION
**Pull request by jonbjarnio**
- Admins can now interact with the attachments submitted in the application-system
- They can both fetch the attachments and override certain attachment type

This PR includes changes in API layer and web project.
- New methods for fetching case attachments and overwriting attachment
- Updated the UI for attachment interactions
- Integration of the attachments onto the case model

image

### CODE REVIEW

File: apps/official-journal-web/src/pages/api/cases/[id]/attachments/[aid]/index.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import type { NextApiRequest, NextApiResponse } from 'next/types'
+import { z } from 'zod'
+import { HandleApiException, LogMethod } from '@dmr.is/decorators'
+import { logger } from '@dmr.is/logging'
+import { isResponse } from '@dmr.is/utils/client'
+
+import { createDmrClient } from '../../../../../../lib/api/createClient'
+import { overrideAttachmentSchema } from '../../../../../../lib/types'
```
Comment (osk) [inline apps/official-journal-web/src/pages/api/cases/[id]/attachments/[aid]/index.ts:8]: fix import

File: apps/official-journal-web/src/pages/api/cases/[id]/attachments/[aid]/index.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        logger.warn(`Failed to fetch attachment`, {
+          error: e,
+          caseId: id,
+          attachmentId: aid,
+        })
+
+        if (isResponse(e)) {
+          const json = await e.json()
+
+          logger.warn(`Failed to fetch attachment ${json.message}`)
```
Comment (osk) [inline apps/official-journal-web/src/pages/api/cases/[id]/attachments/[aid]/index.ts:33]:
```suggestion
          logger.warn(`Failed to fetch attachment`, { message: json.message })
```

File: apps/official-journal-web/src/pages/api/cases/[id]/attachments/[aid]/index.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+          aid,
+          JSON.parse(req.body),
+          req,
+          res,
+        )
+      } catch (e) {
+        if (isResponse(e)) {
+          const json = await e.json()
+
+          logger.warn(`Failed to override attachment ${json.message}`)
```
Comment (osk) [inline apps/official-journal-web/src/pages/api/cases/[id]/attachments/[aid]/index.ts:57]:
```suggestion
          logger.warn(`Failed to override attachment`, { message: json.message })
```

File: libs/shared/utils/src/lib/clientUtils.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@ export const isSingular = (n: number | string): boolean => {
 }
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export const isReponse = (error: any): error is Response => {
+export const isResponse = (error: any): error is Response => {
```
Comment (osk) [inline libs/shared/utils/src/lib/clientUtils.ts:7]:
```suggestion
export const isResponse = (error: unknown): error is Response => {
```

Should be fine and then delete eslint disable

**Review by osk — APPROVED** [review]

File: apps/official-journal-web/src/pages/api/cases/[id]/attachments/[aid]/index.ts (hunk 1 above)
Comment (jonbjarnio) [reply apps/official-journal-web/src/pages/api/cases/[id]/attachments/[aid]/index.ts:8]: Vandamálið er að ég get ekki sett upp root level import þar sem ég mun yfirskrifa tsconfig.base.json paths objectið.

**Pull request merged**